bundles/org.eclipse.jface/src/org/eclipse/jface/util/Throttler.java

```diff
@@ -97,13 +97,27 @@ public Throttler(Display display, Duration minWaitTime, Runnable runnable) {
 	 * returning.
 	 */
 	public void throttledExec() {
+		throttledExec(true);
+	}
+
+	/**
+	 * Like {@link #throttledExec()} but guaranteed to not run the Runnable before
+	 * this method returns.
+	 *
+	 * @since 3.34
+	 */
+	public void throttledAsyncExec() {
+		throttledExec(false);
+	}
+
+	private void throttledExec(boolean allowSyncExec) {
 		if (display.isDisposed()) {
 			return;
 		}
 		if (scheduled.compareAndSet(false, true)) {
 			boolean exception = true;
 			try {
-				if (Thread.currentThread() == display.getThread()) {
+				if (allowSyncExec && Thread.currentThread() == display.getThread()) {
```

Comment:
Somehow the condition and the code do not match. `if(allowSyncExec) timerExec else asyncExec` makes no sense

Reply:
"allow"+"sync" vs "async"

Reply:
`allowSyncExec` -> `allowTimerExec` ? 
Just reading the code for the first time one could assume here is something that is supposed to run synchronous, but it is never the case.

Reply:
I would not mind to rename anything, but actually timerExec is a Runnable that is just run() synchronous when allowSyncExec and runs runner.run() synchronous. Maybe the name "timerExec" does confuse you more

Reply:
`timerExec is a Runnable that is just run() synchronous`

Are we talking about `Display.timerExec()`? It is asynchronous.

Reply:
OK, I see.